src/filepattern/java/FilePattern.java

```diff
@@ -149,106 +174,42 @@ public Pair<HashMap<String, Object>, ArrayList<Path>> next() {
-                BufferedReader buffer = new BufferedReader(infile);
-
-                String line = buffer.readLine();
 
-                if(line.matches("file\\: .+?; corr\\: .+?; position\\: .+?; grid\\: .+?;")) {
-                    return new FilePatternBindings.ExternalVectorPatternObject(builder.path, builder.filePattern); // need to add builder to FPOjbect
-                }
+    private class FilePatternGroupedIterator implements Iterator<Pair<ArrayList<Pair<String, Object>>, ArrayList<Pair<HashMap<String, Object>, ArrayList<Path>>>>> {
 
-                return new FilePatternBindings.ExternalStringPatternObject(builder.path, builder.filePattern, builder.blockSize); // need to add builder to FPOjbect
-            }
+        private int current;
+        private FilePattern fp;
 
-            return new FilePatternBindings.ExternalFilePatternObject(builder.path, builder.blockSize); // need to add builder to FPOjbect
+        public FilePatternGroupedIterator(FilePattern obj) {
+            fp = obj;
+            current = 0;
         }
-    }
-    */
-
-    /*
-    private static class Cast {
-        public ArrayList<Pair<HashMap<String, Object>, ArrayList<Path>>> cast(FilePatternVector vec) {
-            ArrayList<Pair<HashMap<String, Object>, ArrayList<Path>>> casted = new ArrayList<Pair<HashMap<String, Object>, ArrayList<Path>>>();
 
-            for (int i = 0; i < vec.size(); ++i) {
-                FileVector path = vec[i].get1();
-                StringVariant map = vec[i].get0();
-
-            }
+        public boolean hasNext() {
+            return (current + 1) <= this.fp.getSize();
+        }
 
+        
+        public Pair<ArrayList<Pair<String, Object>>, ArrayList<Pair<HashMap<String, Object>, ArrayList<Path>>>> next() {
+            return this.fp.getAtGrouped(current++);
         }
+    
 
     }
-    */
-    public static void main(String[] args) throws IOException {
-        // Pointer objects allocated in Java get deallocated once they become unreachable,
-        // but C++ destructors can still be called in a timely fashion with Pointer.deallocate()
-        //FilePattern fp = new FilePattern("/home/ec2-user/Dev/Demo/test_data/data", "img_r{r:ddd}_c{c:ddd}_{channel:c+}.tif", false, true);
-        //p.init("/home/ec2-user/Dev/Demo/test_data/data", "img_r{r:ddd}_c{c:ddd}_{channel:c+}.tif", false, true);
-        //fp.printFiles();
 
-        //FilePatternVector files = fp.getFiles();
-
-        //ArrayList<Pair<HashMap<String, Object>, ArrayList<Path>>> vec = FilePatternVector.cast(files);
+    public static void main(String[] args) throws IOException {
 
-        FilePattern fp = new FilePattern.FilePatternBuilder("/home/jessemckinzie/Dev/data").recursive(false)
-                                        .filePattern("img_r00{r:d}_c00{c:d}_{channel:c+}.tif")
+        FilePattern fp = new FilePattern.FilePatternBuilder("/Users/jmckinzie/Documents/GitHub/filepattern-1/data").recursive(false)
```

Comment:
local paths leaking? May be do the test in the test framework?

Reply:
I will remove this main class, I was using this for my own testing